### PR TITLE
PWN-8458 - Striga Item Picker

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8458
IMAGE 2023-05-30 19:28:28
IMAGE 2023-05-30 19:28:30
IMAGE 2023-05-30 19:28:32

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/striga/di/StrigaSignupModule.kt [hunk 1]
```diff
@@ -39,12 +40,13 @@ object StrigaSignupModule : InjectionModule {
...
-            StrigaCountryPickerPresenter(
+        factory { (selectedItem: StrigaPickerItem) ->
+            StrigaItemPickerPresenter(
                 strigaOnboardingInteractor = get(),
-                selectedCountry = country
+                selectedItem = selectedItem,
+                strigaElementCellMapper = get()
             )
-        } bind StrigaCountryPickerContract.Presenter::class
+        } bind StrigaItemPickerContact.Presenter::class
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/striga/di/StrigaSignupModule.kt:49]: typo: `StrigaItemPickerCont[r]act`

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt [hunk 2]
```diff
@@ -12,57 +12,57 @@ import org.p2p.uikit.components.finance_block.financeBlockCellDelegate
...
 private const val EXTRA_RESULT = "EXTRA_RESULT"
 private const val EXTRA_SELECTED_COUNTRY = "EXTRA_SELECTED_COUNTRY"
 
-class StrigaCountryPickerFragment : BaseMvpFragment<IView, IPresenter>(R.layout.fragment_striga_country_picker), IView {
+class StrigaItemPickerFragment : BaseMvpFragment<IView, IPresenter>(R.layout.fragment_striga_country_picker), IView {
 
     companion object {
         fun create(
-            selectedCountry: Country?,
+            selectedCountry: StrigaPickerItem,
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt:39]: ~~`selectedCountry`~~ `selectedItem`

File: app/src/main/java/org/p2p/wallet/striga/onboarding/interactor/StrigaOnboardingInteractor.kt [hunk 3]
```diff
@@ -18,4 +20,8 @@ class StrigaOnboardingInteractor(
     }
 
     suspend fun getAllCountries(): List<Country> = countryRepository.getAllCountries()
+
+    fun getOccupationValuesList(): List<StrigaOccupation> = strigaPresetDataLocalRepository.getOccupationValuesList()
```
[the remark this answers is not in the document]
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/striga/onboarding/interactor/StrigaOnboardingInteractor.kt:24]: +1, At least, these 2 methods should be moved to some `StrigaPickerInteractor`

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+class StrigaItemPickerPresenter(
+    private val strigaOnboardingInteractor: StrigaOnboardingInteractor,
+    private val selectedItem: StrigaPickerItem,
+    private val strigaElementCellMapper: StrigaItemCellMapper
+) : BasePresenter<StrigaItemPickerContact.View>(),
+    StrigaItemPickerContact.Presenter {
+
+    private var searchText: String = emptyString()
+    private val searchTextMap = hashMapOf<String, List<FinanceBlockCellModel>>()
```
[the remark this answers is not in the document]
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerPresenter.kt:23]: @gslevinkov Why not use an unsorted map if this map is not iterated anywhere?

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt [hunk 5]
```diff
@@ -12,57 +12,57 @@ import org.p2p.uikit.components.finance_block.financeBlockCellDelegate
...
-        parametersOf(selectedCountry)
+        parametersOf(selectedItem)
     }
     private val binding: FragmentStrigaCountryPickerBinding by viewBinding()
     private val adapter = CommonAnyCellAdapter(
         sectionHeaderCellDelegate(),
         financeBlockCellDelegate { view, _ -> view.setOnClickAction { _, item -> onCountrySelected(item.payload) } },
     )
-    private val selectedCountry: Country? by args(EXTRA_SELECTED_COUNTRY)
+    private val selectedItem: StrigaPickerItem by args(EXTRA_SELECTED_COUNTRY)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt:57]: EXTRA_SELECTED_ITEM

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt [hunk 6]
```diff
@@ -12,57 +12,57 @@ import org.p2p.uikit.components.finance_block.financeBlockCellDelegate
...
     }
 
     override val presenter: IPresenter by inject {
-        parametersOf(selectedCountry)
+        parametersOf(selectedItem)
     }
     private val binding: FragmentStrigaCountryPickerBinding by viewBinding()
     private val adapter = CommonAnyCellAdapter(
         sectionHeaderCellDelegate(),
         financeBlockCellDelegate { view, _ -> view.setOnClickAction { _, item -> onCountrySelected(item.payload) } },
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt:55]: onItemSelected

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt [hunk 7]
```diff
@@ -76,13 +76,8 @@ class StrigaCountryPickerFragment : BaseMvpFragment<IView, IPresenter>(R.layout.
...
     }
 
-    override fun showCountries(items: List<AnyCellItem>) {
-        adapter.items = items
-        binding.containerErrorView.isVisible = items.isEmpty()
-    }
-
     private fun onCountrySelected(selectedItem: Any?) {
-        val selectedCountry = selectedItem as? Country
+        val selectedCountry = selectedItem as? StrigaPickerItem
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt:80]: selectedItem

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt [hunk 8]
```diff
@@ -100,4 +95,13 @@ class StrigaCountryPickerFragment : BaseMvpFragment<IView, IPresenter>(R.layout.
         setTitle(R.string.striga_country)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemPickerFragment.kt:95]: depending on item

File: app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemCellMapper.kt [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            icon = IconWrapperCellModel.SingleEmoji(emoji = item.flagEmoji)
+        )
+        return FinanceBlockCellModel(
+            leftSideCellModel = leftSideCellModel,
+            payload = StrigaPickerItem.CountryItem(item)
+        )
+    }
+
+    fun toUiTitle(value: String): String {
+        return value.replace("_", " ").lowercase().capitalize()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaItemCellMapper.kt:70]: 'capitalize(): String' is deprecated. Use replaceFirstChar instead.

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**